source/guides/guides/test-retries.md

```diff
@@ -0,0 +1,193 @@
+
+```jsx
+{
+  "retries": 1
+}
+```
+
+### Disable Test Retries
+
+If you would like to disable test retries completely (the default configuration), you can use the global configuration to do so by setting `retries` to 0.
+
+```jsx
+{
+  "retries": 0
+}
+```
+
+### Configure specific retries for different modes
+
+If you want to define different retry attempts for `cypress run` versus `cypress open`, you can pass the `retries` option an object with the following options:
+
+- `runMode` allows you to define the number of test retries when running `cypress run`
+- `openMode` allows you to define the number of test retries when running `cypress open`
+
+```jsx
+{
+  "retries": {
+    // Configure retries for `cypress run`
+    // Default is 0
+    "runMode": 2,
+    // Configure retries for `cypress open`
+    // Default is 0
+    "openMode": 1
+  }
+}
+```
+
+## Custom Configurations
+
+### Individual Test(s)
+
+If you want a specific test to run a custom number of retries, you can set this by using the {% url "test's configuration" writing-and-organizing-tests#Test-Configuration %}.
+
+```jsx
+// Customize retries for an individual test
+describe('User sign-up and login', () => {
+  // `it` test block with no custom configuration
+  it('should redirect unauthenticated user to sign-in page', () => {
+    // ...
+  })
+
+  // `it` test block with custom configuration
+  it('allows user to login', {
+    retries: {
+      runMode: 2,
+      openMode: 1
+    }
+  }, () => {
+    // ...
+  })
+})
+```
+
+### Test Suite(s)
+
+If you want a suite of tests to re-run a custom number of times, you can do this by setting the suite's configuration.
+
+```jsx
+// Customizing retries for a suite of tests
+describe('User bank accounts', {
+  retries: {
+    runMode: 2,
+    openMode: 1,
+  }
+}, () => {
+  // Individual tests will be run normally
+  it('allows a user to view their transactions, () => {
+    // ...
+  }
+
+  it('allows a user to edit their transactions, () => {
+    // ...
+  }
+})
+```
+
+You can find more information about custom configurations here: {% url "Test Configuration" configuration#Test-Configuration %}
+
+# Screenshots
+
+With test retries, Cypress will take a separate screenshot for each test attempt. These will be properly labeled with the attempt number appended at the end of each filename.
+
+```js
+// Example file names
+'User Sign-up and Login -- should display login errors.png'
+'User Sign-up and Login -- should display login errors (failed).png'
+'User Sign-up and Login -- should display login errors (attempt 2).png'
+'User Sign-up and Login -- should display login errors (failed) (attempt 2).png'
+'User Sign-up and Login -- should display login errors (attempt 3).png'
+'User Sign-up and Login -- should display login errors (failed) (attempt 3).png'
```

Comment:
I am really confused here - why are there 6 screenshots taken for 3 attempts of a test run? If you're trying to show named screenshots like `cy.screenshot()` versus screenshots taken during failure - this needs to be explicitly clarified. 

**A lot** of our users manipulate the screenshots after being taken and the naming is important to them. This needs to be very clear how we're doing this.